Tests/MapboxDirectionsTests/VisualInstructionComponentTests.swift

```diff
@@ -23,32 +23,126 @@ class VisualInstructionComponentTests: XCTestCase {
+            "name": "us-interstate",
+            "text_color": "black",
+            "display_ref": "242",
+        ]
+        
+        let shieldData = try! JSONSerialization.data(withJSONObject: shieldJSON, options: [])
+        var shield: MapboxShield?
+        XCTAssertNoThrow(shield = try JSONDecoder().decode(MapboxShield.self, from: shieldData))
+        XCTAssertNotNil(shield)
+        let url = URL(string: "https://api.mapbox.com/styles/v1/")
+        if let shield = shield {
+            XCTAssertEqual(shield.baseURL, url)
+            XCTAssertEqual(shield.name, "us-interstate")
+            XCTAssertEqual(shield.textColor, "black")
+            XCTAssertEqual(shield.displayRef, "242")
+        }
+        shield = .init(baseURL: url!, name: "us-interstate", textColor: "black", displayRef: "242")
+        
+        let encoder = JSONEncoder()
+        var encodedData: Data?
+        XCTAssertNoThrow(encodedData = try encoder.encode(shield))
+        XCTAssertNotNil(encodedData)
+        
+        if let encodedData = encodedData {
+            var encodedShieldJSON: [String: Any]?
+            XCTAssertNoThrow(encodedShieldJSON = try JSONSerialization.jsonObject(with: encodedData, options: []) as? [String: Any])
+            XCTAssertNotNil(encodedShieldJSON)
+            
+            XCTAssert(JSONSerialization.objectsAreEqual(shieldJSON, encodedShieldJSON, approximate: false))
+        }
+    }
+    
+    func testShieldImageComponent() {
+        let componentJSON: [String : Any] = [
+            "text": "I 95",
             "type": "icon",
-            "imageBaseURL": "https://s3.amazonaws.com/mapbox/shields/v3/us-42",
+            "mapbox_shield": [
+                "base_url": "https://api.mapbox.com/styles/v1/",
+                "name": "us-interstate",
+                "text_color": "black",
+                "display_ref": "242"
+            ]
         ]
         let componentData = try! JSONSerialization.data(withJSONObject: componentJSON, options: [])
         var component: VisualInstruction.Component?
         XCTAssertNoThrow(component = try JSONDecoder().decode(VisualInstruction.Component.self, from: componentData))
         XCTAssertNotNil(component)
         if let component = component {
             switch component {
-            case .image(let image, let alternativeText):
-                XCTAssertEqual(image.imageBaseURL?.absoluteString, "https://s3.amazonaws.com/mapbox/shields/v3/us-42")
-                XCTAssertEqual(image.imageURL(scale: 1, format: .svg)?.absoluteString, "https://s3.amazonaws.com/mapbox/shields/v3/us-42@1x.svg")
-                XCTAssertEqual(image.imageURL(scale: 3, format: .svg)?.absoluteString, "https://s3.amazonaws.com/mapbox/shields/v3/us-42@3x.svg")
-                XCTAssertEqual(image.imageURL(scale: 3, format: .png)?.absoluteString, "https://s3.amazonaws.com/mapbox/shields/v3/us-42@3x.png")
-                XCTAssertEqual(alternativeText.text, "US 42")
```

Comment:
The Directions API will continue to return `imageBaseURL`, so we need to test that it gets decoded and round-tripped correctly. The `VisualInstruction.Component.imageURL(scale:format:)` case hasn’t been removed, so we should continue to cover the `scale` associated value in this manner.

Reply:
Never mind, the diff obscured the fact that you left the test in place but changed the route number throughout.